lib/ex_unit/test/ex_unit/assertions_test.exs

```diff
@@ -331,12 +331,14 @@ defmodule ExUnit.AssertionsTest do
     rescue
       error in [ExUnit.AssertionError] ->
         """
-        No message matching {:status, ^status} after 0ms.
+        Assertion failed, no matching message after 0ms
         The following variables were pinned:
           status = :valid
-        Process mailbox:
-          {:status, :invalid}\
+        Showing 1 of 1 message in the mailbox\
         """ = error.message
+
+        {pattern, [{:status, :invalid}]} = error.mailbox
```

Comment:
I am not a big fan of asserting the mailbox. The field is an implementation detail and we should probably document it as so.